### PR TITLE
attempt to scaffold out unit testing using vitest

### PR DESCRIPTION
**Pull request by Fbasham**
Add vitest for uniting testing.  

Inspiration taken from remix-run and one of their official stacks:  https://github.com/remix-run/indie-stack

### CODE REVIEW

**Review by gregory-j-baker — COMMENTED** [review]
I'm getting odd behavior when testing this PR.

``` sh
❯ npm run-script typecheck | head -n 25

> typecheck
> tsc

app/entry.server.tsx(58,25): error TS2345: Argument of type 'RouteModules<EntryRouteModule>' is not assignable to parameter of type 'RouteModules'.
  'string' index signatures are incompatible.
    Type 'EntryRouteModule' is not assignable to type 'RouteModule'.
      Types of property 'clientAction' are incompatible.
        Type 'ClientActionFunction | undefined' is not assignable to type 'import("/home/gbaker/Projects/github/dts-stn/canadian-dental-care-plan/frontend/node_modules/@remix-run/react/dist/routeModules").ClientActionFunction | undefined'.
          Type 'ClientActionFunction' is not assignable to type 'import("/home/gbaker/Projects/github/dts-stn/canadian-dental-care-plan/frontend/node_modules/@remix-run/react/dist/routeModules").ClientActionFunction'.
            Types of parameters 'args' and 'args' are incompatible.
              Type 'import("/home/gbaker/Projects/github/dts-stn/canadian-dental-care-plan/frontend/node_modules/@remix-run/react/dist/routeModules").ClientActionFunctionArgs' is not assignable to type 'import("/home/gbaker/Projects/github/dts-stn/canadian-dental-care-plan/frontend/node_modules/@remix-run/server-runtime/dist/routeModules").ClientActionFunctionArgs'.
                Type 'ClientActionFunctionArgs' is not assignable to type '{ serverAction: <T = unknown>() => Promise<SerializeFrom<T>>; }'.
                  The types returned by 'serverAction()' are incompatible between these types.
                    Type 'Promise<import("/home/gbaker/Projects/github/dts-stn/canadian-dental-care-plan/frontend/node_modules/@remix-run/react/node_modules/@remix-run/server-runtime/dist/serialize").SerializeFrom<T>>' is not assignable to type 'Promise<import("/home/gbaker/Projects/github/dts-stn/canadian-dental-care-plan/frontend/node_modules/@remix-run/server-runtime/dist/serialize").SerializeFrom<T>>'.
                      Type 'import("/home/gbaker/Projects/github/dts-stn/canadian-dental-care-plan/frontend/node_modules/@remix-run/react/node_modules/@remix-run/server-runtime/dist/serialize").SerializeFrom<T>' is not assignable to type 'import("/home/gbaker/Projects/github/dts-stn/canadian-dental-care-plan/frontend/node_modules/@remix-run/server-runtime/dist/serialize").SerializeFrom<T>'.
                        Type 'Jsonify<Awaited<T>> | (Parameters<T> extends [ClientLoaderFunctionArgs | ClientActionFunctionArgs] ? unknown : unknown)' is not assignable to type 'SerializeFrom<T>'.
                          Type 'Jsonify<Awaited<T>>' is not assignable to type 'SerializeFrom<T>'.
                            Type 'Awaited<T> extends { toJSON(): infer U; } ? U extends JsonValue ? U : unknown : Awaited<T> extends JsonPrimitive ? JsonPrimitive & Awaited<...> : Awaited<...> extends String ? string : Awaited<...> extends Number ? number : Awaited<...> extends Boolean ? boolean : Awaited<...> extends Promise<...> ? EmptyObject : Aw...' is not assignable to type 'SerializeFrom<T>'.
                              Type 'unknown' is not assignable to type 'SerializeFrom<T>'.
                                Type 'Parameters<T> extends [import("/home/gbaker/Projects/github/dts-stn/canadian-dental-care-plan/frontend/node_modules/@remix-run/react/node_modules/@remix-run/server-runtime/dist/routeModules").ClientLoaderFunctionArgs | import("/home/gbaker/Projects/github/dts-stn/canadian-dental-care-plan/frontend/node_modules/@remi...' is not assignable to type 'Parameters<T> extends [import("/home/gbaker/Projects/github/dts-stn/canadian-dental-care-plan/frontend/node_modules/@remix-run/server-runtime/dist/routeModules").ClientLoaderFunctionArgs | import("/home/gbaker/Projects/github/dts-stn/canadian-dental-care-plan/frontend/node_modules/@remix-run/server-runtime/dist/rout...'.
                                  Type 'SerializeClient<Awaited<Output>> | Serialize<Awaited<Output>>' is not assignable to type 'Parameters<T> extends [ClientLoaderFunctionArgs | ClientActionFunctionArgs] ? SerializeClient<Awaited<Output>> : Serialize<...>'.
                                    Type 'SerializeClient<Awaited<Output>>' is not assignable to type 'Parameters<T> extends [ClientLoaderFunctionArgs | ClientActionFunctionArgs] ? SerializeClient<Awaited<Output>> : Serialize<...>'.
                                      Type '(Awaited<Output> extends TypedResponse<infer U> ? Jsonify<U> : Awaited<Awaited<Output>>) | ({ ...; } & {})' is not assignable to type 'Parameters<T> extends [ClientLoaderFunctionArgs | ClientActionFunctionArgs] ? SerializeClient<Awaited<Output>> : Serialize<...>'.
                                        Type 'Awaited<Output> extends TypedResponse<infer U> ? Jsonify<U> : Awaited<Awaited<Output>>' is not assignable to type 'Parameters<T> extends [ClientLoaderFunctionArgs | ClientActionFunctionArgs] ? SerializeClient<Awaited<Output>> : Serialize<...>'.
```

I tinkered with the branch for a couple of hours and couldn't figure out why this is happening.

I think we should wait to merge this until we can figure this out.

**Comment by gregory-j-baker** [comment]
Also, I acknowledge that this error can be fixed by changing the `RouteModules` import in `entry.server.tsx`, but I don't think that's actually a fix.. more like a workaround.

File: frontend/package.json [hunk 1]
```diff
@@ -56,7 +64,6 @@
       "jsx-a11y"
     ],
     "extends": [
-      "@remix-run/eslint-config",
```
Comment (gregory-j-baker) [inline frontend/package.json:59]: Was this intentionally removed?

File: frontend/vitest.config.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+/// <reference types="vitest" />
+/// <reference types="vite/client" />
```
Comment (gregory-j-baker) [inline frontend/vitest.config.ts:2]: Is this type reference required? I haven't seen it in any vitest examples anywhere.

File: frontend/package.json (hunk 1 above)
Comment (Fbasham) [reply frontend/package.json]: yes (by mistake), and I re-added.

**Review by gregory-j-baker — APPROVED** [review]

**Pull request merged**